openadapt/visualize.py

```diff
@@ -188,70 +189,86 @@ def main():
         ),
     ]
     logger.info(f"{len(action_events)=}")
-    for idx, action_event in enumerate(action_events):
-        if idx == MAX_EVENTS:
-            break
-        image = display_event(action_event)
-        diff = display_event(action_event, diff=True)
-        mask = action_event.screenshot.diff_mask
-
-        if SCRUB:
-            image = scrub.scrub_image(image)
-            diff = scrub.scrub_image(diff)
-            mask = scrub.scrub_image(mask)
-
-        image_utf8 = image2utf8(image)
-        diff_utf8 = image2utf8(diff)
-        mask_utf8 = image2utf8(mask)
-        width, height = image.size
-
-        action_event_dict = row2dict(action_event)
-        window_event_dict = row2dict(action_event.window_event)
-
-        if SCRUB:
-            action_event_dict = scrub.scrub_dict(action_event_dict)
-            window_event_dict = scrub.scrub_dict(window_event_dict)
-
-        rows.append(
-            [
-                row(
-                    Div(
-                        text=f"""
-                        <div class="screenshot">
-                            <img
-                                src="{image_utf8}"
-                                style="
-                                    aspect-ratio: {width}/{height};
-                                "
-                            >
-                            <img
-                                src="{diff_utf8}"
-                                style="
-                                    aspect-ratio: {width}/{height};
-                                "
-                            >
-                            <img
-                                src="{mask_utf8}"
-                                style="
-                                    aspect-ratio: {width}/{height};
-                                "
-                            >
-                        </div>
-                        <table>
-                            {dict2html(window_event_dict , None)}
-                        </table>
-                    """,
-                    ),
-                    Div(
-                        text=f"""
-                        <table>
-                            {dict2html(action_event_dict)}
-                        </table>
-                    """
+
+    rows = []
+    num_events = (
+        min(MAX_EVENTS, len(action_events))
+        if MAX_EVENTS is not None
+        else len(action_events)
+    )
+    with tqdm(
+            total=num_left,
```

Comment:
undefined, I think it should be:

```suggestion
            total=num_events,
```

Reply:
I named it `num_left` as It denotes the number of Event "**_left_**" to be written in database.

num_event is a better choice though. :)

A commit will follow

Reply:
Addressed in: [2c26a45](https://github.com/MLDSAI/OpenAdapt/pull/318/commits/2c26a459e8103ea64580dce2ce7ef5ca544eee34)